include/memory_utils.h

```diff
@@ -0,0 +1,9 @@
+#ifndef MEMORY_UTILS_H
+#define MEMORY_UTILS_H
```

Comment:
I wouldn't mind if this were just called "memory.h" tbh, I don't think "utils" really adds much

Reply:
its what we decided to name the C file that holds these two functions. we could rename both, but I think it may make it sound more important than it is. dont have too strong an opinion otherwise